test/cli/provider_test.go

```diff
@@ -512,5 +532,37 @@ func TestProvider(t *testing.T) {
 		// Should be a non-negative integer (0 or positive)
 		assert.GreaterOrEqual(t, result, 0)
 	})
+}
+
+func TestProvider(t *testing.T) {
+	t.Parallel()
 
+	variants := []struct {
+		name      string
+		reprovide bool
+		apply     cfgApplier
+	}{
+		{
+			name:      "BurstProvider",
+			reprovide: true,
+			apply: func(n *harness.Node) {
+				n.SetIPFSConfig("Reprovider.Sweep.Enabled", false)
+			},
+		},
+		{
+			name:      "SweepingProvider",
+			reprovide: false,
+			apply: func(n *harness.Node) {
+				n.SetIPFSConfig("Reprovider.Sweep.Enabled", true)
+			},
+		},
```

Comment:
Note: I tried adding tests with `Routing.AcceleratedDHTClient` set to true, but they are timing out. I didn't investigate why since it seems out of the scope of this PR to add/debug tests related to the Accelerated DHT client.